### PR TITLE
copr: add quarter function

### PR DESCRIPTION
**Pull request by AT-Fieldless**
Hi! I'm new to the project. this is my small PR about TiDB function migration.

What problem does this PR solve?
Issue Number: ref https://github.com/tikv/tikv/issues/5751
Problem Summary: migrating quarter from TiDB

What is changed and how it works?
What's Changed:

Implement quarter function.

Related changes
No Related Changes
Check List
Tests

Unit test
### Release note 

```release-note
Implement RPN function quarter
```

### CODE REVIEW

**Review by skyzh — COMMENTED** [review]
Please follow the contribution guide and fix fmt/ clippy, thanks!

File: components/tidb_query_expr/src/impl_time.rs [hunk 1]
```diff
@@ -865,6 +865,12 @@ pub fn duration_string_time_diff(
     duration_duration_time_diff(ctx, arg1, &arg2)
 }
 
+#[rpn_fn(nullable)]
+#[inline]
+pub fn quater(t: Option<&DateTime>) -> Result<Option<Int>> {
```
Comment (sticnarf) [inline components/tidb_query_expr/src/impl_time.rs:870]: Typo: quater -> quarter

File: components/tidb_query_expr/src/impl_time.rs [hunk 2]
```diff
@@ -865,6 +865,12 @@ pub fn duration_string_time_diff(
     duration_duration_time_diff(ctx, arg1, &arg2)
 }
 
+#[rpn_fn(nullable)]
```
Comment (sticnarf) [inline components/tidb_query_expr/src/impl_time.rs:868]: I think it's ok to just use `#[rpn_fn]` and let this function accepts an `&DateTime`.

**Review by sticnarf — COMMENTED** [review]
The rest LGTM.

**Review by skyzh — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]

**Pull request merged**